src/views/Onboarding/OnboardingSetup.vue

```diff
@@ -63,7 +63,11 @@ export default {
     async confirmMnemonic () {
       this.currentStep = 'congrats'
       await this.setupWallet({ key: this.password })
-      await this.createWallet({ key: this.password, mnemonic: this.mnemonic }) // mnemonic prop can be null to generate new seed
+      await this.createWallet({
+        key: this.password,
+        mnemonic: this.mnemonic,
+        imported: !!this.imported
```

Comment:
You could also just check that mnemonic is null as indicated in the comment below. 
Although i think `imported` prop is much cleaner

Reply:
Agreed with you fixed and checking now if we have the mnemonic or not and set the imported flag